src/mlagility/cli/report.py

```diff
@@ -1,521 +1,216 @@
-            measured_groq_power="TODO",
-            measured_gpu_latency="TODO",
-            measured_gpu_throughput="TODO",
-            result=result,
-            build_duration="TODO",
-            compiler_duration="TODO",
-            assembler_duration="TODO",
-        )
-
-    # results_csv.write_legend()
-    # results_csv.write_headers()
-    results_csv.write(spreadsheet_file)
+    # Description: Task being performed by the model
+    # Source: Specified as part of the .py file using the label "task"
+    task: str = "-"
 
-    # Print message with the output file path
-    printing.log("Summary spreadsheet saved at ")
-    printing.logn(str(spreadsheet_file), printing.Colors.OKGREEN)
+    # Description: Number of chips needed to run the model when mapped to Groq
+    # Source: Computed by GroqFlow
+    groq_chips_used: str = "-"
 
-    # Make the logs.yaml file
-    with open(results_file, "w", encoding="utf8") as outfile:
-        yaml.dump(results_dict, outfile)
+    # Description: Number of chips needed to run the model when mapped to Groq
+    # Source: Computed by GroqFlow
+    groq_chips_used: str = "-"
 
-    # Print message with the output file path
-    printing.log("Build results across all models collected at ")
-    printing.logn(str(results_file), printing.Colors.OKGREEN)
+    # Description: An estimation of latency, in ms, to run one invocation of the
+    #              model on a GroqNode server (includes I/O).
+    # Source: computed by GroqFlow
+    groq_estimated_latency: str = "-"
 
+    # Description: Mean total latency, in ms, to run one invocation using TensorRT.
+    # Source: benchit.benchmark() on 100 runs
+    nvidia_latency: str = "-"
 
-def op_spreadsheet(cache_dir):
+    # Description: Mean total latency, in ms, to run one invocation using OnnxRuntime.
+    # Source: benchit.benchmark() on 100 runs
+    x86_latency: str = "-"
 
-    # Declare file paths
-    spreadsheet_file = os.path.join(cache_dir, op_support_filename)
 
-    with open(spreadsheet_file, "w", newline="", encoding="utf8") as spreadsheet:
-        writer = csv.writer(spreadsheet)
+def summary_spreadsheet(args) -> None:
+
+    # Input arguments from CLI
+    cache_dirs = [os.path.expanduser(dir) for dir in args.cache_dirs]
+    report_dir = os.path.expanduser(args.report_dir)
+
+    # Name report file
+    report_path = os.path.join(report_dir, get_report_name())
+
+    # Create report dict
+    Path(report_dir).mkdir(parents=True, exist_ok=True)
+    report = {}
 
-        # Git commit placeholder
-        writer.writerow(
-            [
-                "Git Commit: ",
-            ]
-        )
+    # Add results from all cache folders
+    for cache_dir in cache_dirs:
 
-        writer.writerow(
-            [
-                "Model Name",
-                "Unsupported ops (Optimized ONNX)",
-            ]
-        )
+        # Check cache directory
+        check_cache_dir(cache_dir)
```

Comment:
Nice! 

The comment seems unnecessary though since its word-for-word the same as the function name. Either add more detail to the comment or remove it.

Reply:
Added more detailed comment